### DISCUSSION TITLE
AntiFlash & EarRinging Patch

### DISCUSSION THREAD

**Discussion by D1GQ** [winstxnhdw/lc-hax#438] [Ideas]
# Overview 
As the title says, made a patch that prevents StunGrade's and FlashBang's from making your ears ring and making your screen white.

## AntiFlash&EarRingingPatch.cs
```
using HarmonyLib;

class AntiFlashAndEarRingingPatch {
    [HarmonyPatch(typeof(HUDManager), "Update")]
    class AntiFlashPatch {
        [HarmonyPrefix]
        static bool Prefix(HUDManager __instance) {
            __instance.flashFilter = 0f;
            return true;
        }
    }

    [HarmonyPatch(typeof(SoundManager), "SetEarsRinging")]
    class AntiEarRingingPatch {
        [HarmonyPrefix]
        static bool Prefix() {
            return false;
        }
    }
}
```

https://github.com/winstxnhdw/lc-hax/discussions/339#discussion-6164561

**Comment by Dexteriority** [comment]
Does this also patch Shotgun Blast ringing?

**Reply by winstxnhdw** [reply]
Probably.

**Reply by Dexteriority** [reply]
Its already inside the ShotgunPatch no?
It should be fine if there is also another patch doing it right?

**Reply by winstxnhdw** [reply]
Good catch.
